File: src/db/schema/azkar.ts
// Wave15.4: stripped 19 lines of leading LLM narration prose
import { pgTable, serial, text, timestamp, boolean } from 'drizzle-orm/pg-core';
import { sql } from 'drizzle-orm';

export const azkarTable = pgTable('azkar', {
  id: serial('id').primaryKey(),
  title: text('title').notNull(),
  content: text('content').notNull(),
  createdAt: timestamp('created_at').defaultNow(),
  updatedAt: timestamp('updated_at').defaultNow(),
  isFavorite: boolean('is_favorite').default(false),
});

export type Azkar = typeof azkarTable.$inferSelect;
export type NewAzkar = typeof azkarTable.$inferInsert;

export const azkar = pgTable('azkar', {
  id: uuid("id").primaryKey().defaultRandom(),
});

File: android/app/src/main/java/MawaqitMasjid/data/Azkar.kt
// Wave15.4: stripped 2 lines of leading LLM narration prose
/** Domain model for "Azkar" with stable identity and a creation
 *  timestamp. */
data class Azkar(
    val id: String,
    val name: String,
    val createdAtEpochMs: Long = System.currentTimeMillis(),
)
